pkg/daemon/controller/controller.go

```diff
@@ -310,8 +316,7 @@ func (sc *ServiceController) reconcileEnvironmentVariables(envVars map[string]st
 	if err != nil {
 		return true, fmt.Errorf("error waiting for environment vars to get picked up by processes: %w", err)
 	}
-	// Remove the reboot annotation after we know the reboot occurred successfully. No-op if already not present
-	return false, metadata.RemoveRebootAnnotation(sc.ctx, sc.client, node)
+	return false, nil
```

Comment:
sorry if there was a miscommunication. I was saying you could leave this as WaitForRebootAnnotation to be removed

Reply:
I am confused again :laughing: 
Isn't adding the reboot anno to the predicate accomplishing the same thing as waiting for the anno to be removed? We'd only reach this point in the code if the anno is not present already (otherwise we wouldn't even reconcile)

Reply:
if you just continue wouldn't this result in the node getting marked as configured before it is rebooted (and working properly)?

thats why i thought the wait was necessary

Reply:
That was true before the addition to the predicate.

Now, we are returning `false, nil` here only if the node is safe to be configured. All of the following must be true to hit this line
- there is no reboot anno on the node (reconcile event actually get processed)
- system level env vars are properly set (otherwise function would exit early after applying RebootRequired anno)
- process level env vars are picked up properly (otherwise the Poll would error out)
That's a green light to continue on with the services configuration etc

Reply:
Thanks for the explanation, makes sense to me